common-http/src/main/java/io/cdap/common/http/HttpResponse.java

```diff
@@ -69,6 +102,28 @@ public String getResponseBodyAsString(Charset charset) {
     return headers;
   }
 
+  public void consumeContent() throws IOException {
+    if (inputStream == null) {
+      conn.disconnect();
+      consumer.onFinished();
+      return;
+    }
+
+    try (ReadableByteChannel channel = Channels.newChannel(inputStream)) {
+      ByteBuffer buffer = ByteBuffer.allocate(consumer.getChunkSize());
+      while (channel.read(buffer) >= 0) {
+        boolean continueReading = consumer.onReceived(buffer);
```

Comment:
I think you should flip the buffer before provider to the consumer. It is unnatural to require the consumer implementation to flip it.

Reply:
done